src/NLog/Targets/FileTarget.cs

```diff
@@ -390,16 +390,14 @@ public string ArchiveNumbering
             get => _archiveNumbering ?? "Sequence";
             set
             {
-                if (string.Equals(value, _archiveNumbering))
-                    return;
-
-                _archiveNumbering = string.IsNullOrEmpty(value) ? null : value.Trim();
-                if (_archiveNumbering is null || string.IsNullOrEmpty(_archiveNumbering))
+                var archiveNumbering = value?.Trim() ?? string.Empty;
+                if (string.IsNullOrEmpty(archiveNumbering) || string.Equals(_archiveNumbering, archiveNumbering, StringComparison.OrdinalIgnoreCase))
```

Comment:
```suggestion
                if (string.IsNullOrEmpty(archiveNumbering) || archiveNumbering.Equals(_archiveNumbering, StringComparison.OrdinalIgnoreCase))
```
this will reduce one null check that is called in the static string.Equals method

Reply:
Old habit of using `string.Equals` not that worried about a bonus null-check for an obsolete property. But thank you for noticing.